can't be used with the starlight 0.34.0

It seems that the plugin can't work with the newest starlight

15:45:08 [ERROR] [@astrojs/starlight] An unhandled error occurred while running the "astro:config:setup" hook
path2.endsWith is not a function
  Location:
    E:/chasezhj/chasezhj/node_modules/.pnpm/starlight-site-graph@0.3.0_@astrojs+starlight@0.34.0_astro@5.6.1_@netlify+blobs@8.1.1_jiti@2._raj55mfepzl47fhzbssghxaxu4/node_modules/starlight-site-graph/sitemap/util.ts:21:16
  Stack trace:
    at ensureTrailingSlash (E:/chasezhj/chasezhj/node_modules/.pnpm/starlight-site-graph@0.3.0_@astrojs+starlight@0.34.0_astro@5.6.1_@netlify+blobs@8.1.1_jiti@2._raj55mfepzl47fhzbssghxaxu4/node_modules/starlight-site-graph/sitemap/util.ts:21:16)
    at file:///E:/chasezhj/chasezhj/node_modules/.pnpm/zod@3.24.2/node_modules/zod/lib/index.mjs:3517:40
    at async astro:config:setup (E:/chasezhj/chasezhj/node_modules/.pnpm/@astrojs+starlight@0.34.0_astro@5.6.1_@netlify+blobs@8.1.1_jiti@2.4.2_rollup@4.34.9_typescript@5.8.2_yaml@2.7.0_/node_modules/@astrojs/starlight/index.ts:43:30)
    at async runHookInternal (file:///E:/chasezhj/chasezhj/node_modules/.pnpm/astro@5.6.1_@netlify+blobs@8.1.1_jiti@2.4.2_rollup@4.34.9_typescript@5.8.2_yaml@2.7.0/node_modules/astro/dist/integrations/hooks.js:51:5)
    at async createContainer (file:///E:/chasezhj/chasezhj/node_modules/.pnpm/astro@5.6.1_@netlify+blobs@8.1.1_jiti@2.4.2_rollup@4.34.9_typescript@5.8.2_yaml@2.7.0/node_modules/astro/dist/core/dev/container.js:23:14)
 ELIFECYCLE  Command failed with exit code 1.

--- a/packages/starlight-site-graph/index.ts
+++ b/packages/starlight-site-graph/index.ts
@@ -29,7 +29,7 @@ export default function plugin(userConfig?: StarlightSiteGraphConfig): Starlight
 					}
 
 					for (const link of Object.values(config.social ?? {})) {
-						starlightIgnoredLinks.push(`!${ensureTrailingSlash(link, addTrailingSlashes)}`);
+						starlightIgnoredLinks.push(`!${ensureTrailingSlash(link.href, addTrailingSlashes)}`);
 					}
 
 					parsedConfig.sitemapConfig.linkInclusionRules.splice(-1, 0, ...starlightIgnoredLinks);

